bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+        if (agentResponse.getStatus() != 200) {
+            throw new MercedesMeAuthException("Failed to post user agent. HTTP " + agentResponse.getStatus());
+        }
+    }
+
+    /**
+     * Send user name.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private void sendUsername(HttpClient loginHttpClient) throws MercedesMeAuthException, MercedesMeApiException {
+        Request userRequest = loginHttpClient.POST(baseUrl + "/ciam/auth/login/user");
+        userRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        userRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        userRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
+        userRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        userRequest.header(HttpHeader.ORIGIN, baseUrl);
+        userRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");
+
+        JSONObject userContent = new JSONObject();
+        userContent.put("username", config.email);
+        userRequest.content(new StringContentProvider(userContent.toString(), "utf-8"));
+
+        ContentResponse userResponse = send(userRequest);
+        int status = userResponse.getStatus();
+        logger.trace("Step 3: Post username {} - {}", status, userResponse.getContentAsString());
+        if (status != 200) {
+            throw new MercedesMeAuthException("Failed to post username " + config.email + ". HTTP " + status);
+        }
+    }
+
+    /**
+     * Perform login with user name and password to get pre-login token.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct or token not delivered
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private String performPasswordLogin(HttpClient loginHttpClient)
+            throws MercedesMeAuthException, MercedesMeApiException {
+        Request loginRequest = loginHttpClient.POST(baseUrl + "/ciam/auth/login/pass");
+        loginRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        loginRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        loginRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
+        loginRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        loginRequest.header(HttpHeader.ORIGIN, baseUrl);
+        loginRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");
+
+        String rid = generateCodeVerifier(24);
+        JSONObject loginContent = new JSONObject();
+        loginContent.put("username", config.email);
+        loginContent.put("password", config.password);
+        loginContent.put("rememberMe", false);
+        loginContent.put("rid", rid);
+        loginRequest.content(new StringContentProvider(loginContent.toString(), "utf-8"));
```

Comment:
See comment further down.
```suggestion
        loginRequest.agent(Constants.AUTH_USER_AGENT);
        loginRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
        loginRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
        loginRequest.header(HttpHeader.ORIGIN, baseUrl);
        loginRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");

        String rid = generateCodeVerifier(24);
        JSONObject loginContent = new JSONObject();
        loginContent.put("username", config.email);
        loginContent.put("password", config.password);
        loginContent.put("rememberMe", false);
        loginContent.put("rid", rid);
        loginRequest.content(new StringContentProvider(loginContent.toString(), "utf-8"), CONTENT_TYPE_JSON);
```